networks/xml_cnn.py

```diff
@@ -0,0 +1,71 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from networks.base import BaseModel
+
+
+def out_size(l_in, kernel_size, channels, padding=0, dilation=1, stride=1):
+    a = l_in + 2 * padding - dilation * (kernel_size - 1) - 1
+    b = int(a / stride)
```

Comment:
since we have l_in, we can have l_out
and
l_out = math.floor(a/stride)+1
we can then return
l_out * channels
BTW I prefer floor over int for better readability
Also can add a comment to refer to
https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html
about the l_out calculation

Reply:
Solved.